src/Microsoft.DotNet.ProjectModel/Graph/IMergeable.cs

```diff
@@ -0,0 +1,7 @@
+namespace Microsoft.DotNet.ProjectModel.Graph
+{
+    public interface IMergeable<T>
+    {
+        void MergeWith(T m);
```

Comment:
Should this just be MergeWithFragment, to reduce the scope of only merging with fragments?